src/enforester.js

```diff
@@ -1314,17 +1327,26 @@ export class Enforester {
   enforestNewExpression() {
     this.matchKeyword('new');
     let callee;
-    if (this.isKeyword(this.peek(), 'new')) {
+    let lookahead = this.peek();
+    if (this.isKeyword(lookahead, 'new')) {
       callee = this.enforestNewExpression();
-    } else if (this.isKeyword(this.peek(), 'super')) {
+    } else if (this.isKeyword(lookahead, 'super')) {
+      callee = this.enforestLeftHandSideExpression({ allowCall: false });
+    } else if (this.isParens(lookahead) || this.isBrackets(lookahead) || this.isBraces(lookahead) || this.isTemplate(lookahead)) {
       callee = this.enforestExpressionLoop();
-    } else if (this.isPunctuator(this.peek(), '.') && this.isIdentifier(this.peek(1), 'target')) {
+    } else if (this.isPunctuator(lookahead, '.') && this.isIdentifier(this.peek(1), 'target')) {
       this.advance();
       this.advance();
       return new Term('NewTargetExpression', {});
     } else {
       callee = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
     }
+
+    if(this.isTemplate(this.peek())) {
+      this.term = callee;
+      callee = this.enforestLeftHandSideExpression({ allowCall: false });
+    }
+
```

Comment:
Besides `new.target` all these cases can be handled in `enforestLeftHandSideExpression` right?